src/sync/pull.ts

```diff
@@ -427,18 +438,18 @@ async function addChangedKeysForIndexes(
       const newIndex = newIndexes.get(oldIndexName);
       if (newIndex !== undefined) {
         const changedKeys = await newIndex.withMap(read, async newMap => {
-          return btree.changedKeys(oldMap, newMap);
+          return btree.diff(oldMap, newMap);
         });
 
         newIndexes.delete(oldIndexName);
         if (changedKeys.length > 0) {
-          changedKeysMap.set(oldIndexName, changedKeys);
+          diffsMap.set(oldIndexName, changedKeys);
         }
       } else {
-        // old index name is not in the new indexes. All keys changed!
-        const changedKeys = await allKeys(oldMap);
+        // old index name is not in the new indexes. All entries changed!
+        const changedKeys = await diffForAll(oldMap);
```

Comment:
This needs to be all deleted, not all added, correct?  I think diffForAll need to take an op.

Reply:
Seems we probably need a test for this.

Reply:
I see what you are saying now... Will look